optimum/exporters/onnx/__main__.py

```diff
@@ -353,6 +357,10 @@ def main_export(
                 f"Asked to export a {model_type} model for the task {task}{autodetected_message}, but the Optimum ONNX exporter only supports the tasks {', '.join(model_tasks.keys())} for {model_type}. Please use a supported task. Please open an issue at https://github.com/huggingface/optimum/issues if you would like the task {task} to be supported in the ONNX export for {model_type}."
             )
 
+        # TODO: Fix in Transformers so that SdpaAttention class can be exported to ONNX. `attn_implementation` is introduced in Transformers 4.36.
+        if model_type in SDPA_ARCHS_ONNX_EXPORT_NOT_SUPPORTED and _transformers_version >= version.parse("4.35.99"):
+            loading_kwargs["attn_implementation"] = "eager"
```

Comment:
Bart decoder and Whisper encoder + decoder should be fixable since the query length is static there (so no need to actually have a controlflow on it).

And for bart encoder, we are always non-causal no matter q_len so we could just rely on that.